### PR TITLE
View joins with any and all not working correctly

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Fix filtering issues with relationship filtering.

### CODE REVIEW

**Comment by aptkingston** [comment]
I assume this would fix https://linear.app/budibase/issue/BUDI-8738/filter-by-relationship-not-working-on-views?

File: packages/server/src/api/routes/tests/search.spec.ts [hunk 1]
```diff
@@ -2329,6 +2330,211 @@ describe.each([
...
+                            },
+                          ],
+                        },
+                      },
+                    ],
+                  },
+                }).toContainExactly([
+                  { name: "foo", productCat: [{ _id: productCatRows[0]._id }] },
+                  { name: "bar", productCat: [{ _id: productCatRows[1]._id }] },
+                  // { name: "baz", productCat: undefined }, // TODO
```
Comment (adrinr) [inline packages/server/src/api/routes/tests/search.spec.ts:2509]: This is the current behaviour and fixing this required quite a lot of changes. Doing this in a different branch while investigating the best approach

**Comment by adrinr** [comment]
> I assume this would fix https://linear.app/budibase/issue/BUDI-8738/filter-by-relationship-not-working-on-views?

Yes, this should fix the same issue

**Review by samwho — APPROVED** [review]
The tests 👨‍🍳 💋

**Pull request merged**